.github/workflows/build.yml

```diff
@@ -0,0 +1,160 @@
+        run: |
+          curl -L https://sourceforge.net/projects/sbml/files/libsbml/win-dependencies/libSBML_dependencies_vs15_release_x64_static.zip/download > dependencies.zip
+          unzip dependencies.zip -d dependencies
+          cp -r dependencies/libSBML\ Dependencies-1.0.0-b1-win64/* dependencies
+          rm -r dependencies/libSBML*
+          curl -L https://prdownloads.sourceforge.net/swig/swigwin-3.0.12.zip > swig.zip
+          unzip -o swig.zip -d swig
+      - name: setup Windows environment
+        # this is separate from the SWIG download itself, because it needs to be added to the path also when SWIG is cached
+        if: matrix.platform == 'windows-latest'
+        shell: bash
+        run: |
+          echo $GITHUB_WORKSPACE"/swig/swigwin-3.0.12/" >> $GITHUB_PATH
+          echo RUNTIME_LINKING_OPTION="-DWITH_STATIC_RUNTIME=ON" >> "${GITHUB_ENV}"
+      - name: Install Ubuntu dependencies
+        # ubuntu already has SWIG and libxml2 by default
+        if: matrix.platform == 'ubuntu-16.04'
+        shell: bash
+        run: |
+          sudo apt-get install -y check libxerces-c-dev expat ccache
+          echo PYTHON_LINKING_OPTION="-DPYTHON_USE_DYNAMIC_LOOKUP=ON" >> "${GITHUB_ENV}"
+      - name: Install MacOS dependencies
+        # MacOS already has libxml2 by default
+        if: matrix.platform == 'macos-latest'
+        shell: bash
+        run: |
+          brew install check swig xerces-c expat ccache
+          echo PYTHON_LINKING_OPTION="-DPYTHON_USE_DYNAMIC_LOOKUP=ON" >> "${GITHUB_ENV}"
+      ### setup ccache, not on Windows ###
+      - name: Prepare ccache timestamp
+        if: matrix.platform != 'windows-latest'
+        id: ccache_cache_timestamp
+        shell: cmake -P {0}
+        run: |
+          string(TIMESTAMP current_date "%Y-%m-%d-%H;%M;%S" UTC)
+          message("::set-output name=timestamp::${current_date}")
+      - name: Set ccache cache directory
+        if: matrix.platform != 'windows-latest'
+        shell: bash
+        run: |
+          echo "CCACHE_DIR=${{runner.workspace}}/.ccache" >> "${GITHUB_ENV}"
+          echo "COMPILER_LAUNCHER=ccache" >> "${GITHUB_ENV}"
+      - name: cache ccache files
+        if: matrix.platform != 'windows-latest'
+        uses: actions/cache@v2
+        with:
+          path: ${{runner.workspace}}/.ccache
+          key:
+            ${{ runner.os }}-test-${{ matrix.xml_parser_option }}-${{
+            matrix.with_namespace }}-${{ matrix.strict }}-${{
+            steps.ccache_cache_timestamp.outputs.timestamp }}
+          restore-keys: |
+            ${{ runner.os }}-test-${{ matrix.xml_parser_option }}-${{ matrix.with_namespace }}-${{ matrix.strict }}-${{ steps.ccache_cache_timestamp.outputs.timestamp }}
+            ${{ runner.os }}-test-${{ matrix.xml_parser_option }}-${{ matrix.with_namespace }}-${{ matrix.strict }}-
+            ${{ runner.os }}-test-${{ matrix.xml_parser_option }}-${{ matrix.with_namespace }}-${{ matrix.strict }}-
```

Comment:
Forgive my ignorance, but what are these identical lines doing?

Reply:
This is a relic from when I tried to use `ccache` that got copied over. It's a non-critical bug that has been on my todo for a while to remove/improve on. Can be removed here, I think.